### DISCUSSION TITLE
More approciable way to test master VIMRUNTIME changes

### DISCUSSION THREAD

**Discussion by saccarosium** [vim/vim#15923] [Q&A] [answered]
TLDR: I want a way to runtime patches locally without installing it globally

I've been testing a lot more patches to the runtime recently. My apprach is to go to the PR branch and then do `sudo make install` this becomes annoying really quickly. I've also tried to change `runtimepath` to the local runtime but without luck.

**Comment by chrisbra** [comment, marked as answer]
what I usually do is `export $VIMRUNTIME=/home/cb/code/vim/runtime`

**Reply by Konfekt** [reply]
That's the best solution. I've just linked those files of interest into ~/plugin/ say, loading first and overriding those in runtime:

```
# doc/options.txt (lines 6812-6816)
	You probably should always include $VIMRUNTIME somewhere, to use the
	distributed runtime files.  You can put a directory before $VIMRUNTIME
	to find files which replace a distributed runtime files. 
```

**Comment by ychin** [comment]
As long as your repository is cloned to a folder named "vim", just do the following when you run `configure` in the root directory:

```
./configure --datarootdir=`pwd`/.. <other_configure_arguments>
```

This will configure Vim to properly use the runtime folder in your current repository, and you just need to build Vim with `make` and it will work as you expected. I find this much easier than having to export a `VIMRUNTIME` environmental variable which is kind of annoying and error prone.

If your repository is *not* named "vim", then I prefer just calling `make VIMRUNTIMEDIR=./src/runtime` instead of setting the env var because it's more transferable across different directories, and the compiled Vim will have the correct runtime directory baked into the executable and can be called from anywhere even without it set.

---

Note: The configure command works because in the Makefile we have `VIMLOC = $(DATADIR)$(VIMDIR)`, where `VIMDIR` is defined as "vim". It's kind of a hack in a way but it works.

Note 2: I think setting an env var `VIMRUNTIME` is error prone is that now you are making a global variable that would change the behavior of a vanilla system installed Vim as well. Generally I only want the Vim under local iteration to use the runtime folders in my development directory. I want my global Vim to always use the installed runtime files.

I sometimes wonder if the configure script should really have a developer / local repository mode. Currently there's no good ergonomic way to just say "I want to iterate on the current repository".
